src/main/java/org/terasology/las/bases/BaseProvider.java

```diff
@@ -43,12 +48,22 @@ public void setSeed(long seed) {
 
     @Override
     public void process(GeneratingRegion region) {
+        SurfacesFacet surfacesFacet = region.getRegionFacet(SurfacesFacet.class);
         Border3D border = region.getBorderForFacet(BaseFacet.class);
         BaseFacet facet = new BaseFacet(region.getRegion(), border);
 
         for (Base base : fixedBases) {
             facet.add(base);
         }
+
+        BlockRegion surfaceRegion = surfacesFacet.getWorldRegion();
+        for (Vector3ic pos : surfaceRegion) {
```

Comment:
The base regions are a lot smaller, so I think it would make more sense to loop over those and then check if it's in the generating region. Maybe:

```suggestion
        for (Vector3ic pos : Iterables.concat(redBaseRegion, blackBaseRegion)) {
            if (surfaceRegion.contains(pos)) {
```